nsxt/resource_nsxt_upgrade_run.go

```diff
@@ -503,9 +523,83 @@ func updateUpgradeUnitGroups(upgradeClientSet *upgradeClientSet, d *schema.Resou
 	for _, groupI := range d.Get(componentToGroupKey[component]).([]interface{}) {
```

Comment:
Looks like the code doesn't delete custom groups that were created with previous apply (which might have failed) but are not present in current config, is that correct?
To solve this, `d.GetChange("host_group")` might be useful